### DISCUSSION TITLE
sample/drivers/adc 3.2.x compatibility break

### DISCUSSION THREAD

**Discussion by rjmaris** [zephyrproject-rtos/zephyr#56437] [General]
I have tried to 'restore' the v3.1.0 console view of adc label followed by channel index (instead of the new adc register address with channel index).

In the docs it is stated:
```
DT_LABEL(node_id)¶
Equivalent to DT_PROP(node_id, label)

Deprecated:
Use DT_PROP(node_id, label)
```
Well, what exactly should I insert where the four asterisks are? - precisely in this fragment of sample/driver/adc/main.c (which I'm taking from v3.1.0 - I already replaced DT_LABEL by DT_PROP):
```
#define LABEL_AND_COMMA(node_id, prop, idx) \
	DT_PROP(DT_IO_CHANNELS_CTLR_BY_IDX(node_id, idx), ****),

/* Labels of ADC controllers referenced by the above io-channels. */
static const char *const adc_labels[] = {
	DT_FOREACH_PROP_ELEM(DT_PATH(zephyr_user), io_channels,
			     LABEL_AND_COMMA)
};
```
This macro&code snippet is prepended by the following snippet that constructs an array with channel id's:
```
#define DT_SPEC_AND_COMMA(node_id, prop, idx) \
	ADC_DT_SPEC_GET_BY_IDX(node_id, idx),

/* Data of ADC io-channels specified in devicetree. */
static const struct adc_dt_spec adc_channels[] = {
	DT_FOREACH_PROP_ELEM(DT_PATH(zephyr_user), io_channels,
			     DT_SPEC_AND_COMMA)
};
```

**Comment by yashi** [comment]
Literally `label`.

That's how you get the "label" property of a node.

**Comment by rjmaris** [comment]
@yashi After inspection of devicetree.h, I see that you're right. The only difference related to v3.1.0 is that it is now explicitly marked as deprecated macro.
That means: something else causes a build error.
